e2e/nomostest/clusters/gke.go

```diff
@@ -58,6 +59,71 @@ func withKubeConfig(cmd *exec.Cmd, kubeconfig string) *exec.Cmd {
 	return cmd
 }
 
+// listOperations lists RUNNING operations on the target cluster
+func listOperations(t testing.NTB, name string) ([]string, error) {
+	args := []string{
+		"container", "operations", "list",
+		"--project", *e2e.GCPProject,
+		"--filter", fmt.Sprintf("status = RUNNING AND targetLink ~ %s", name),
+		"--format", "value(name)",
+	}
+	if *e2e.GCPZone != "" {
+		args = append(args, "--zone", *e2e.GCPZone)
+	}
+	if *e2e.GCPRegion != "" {
+		args = append(args, "--region", *e2e.GCPRegion)
+	}
+	t.Logf("gcloud %s", strings.Join(args, " "))
+	cmd := exec.Command("gcloud", args...)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, errors.Errorf("failed to list operations for %s: %v\nstdout/stderr:\n%s",
+			name, err, string(out))
+	}
+	operations := strings.Fields(string(out))
+	return operations, nil
+}
+
+// waitOperation waits for the provided operation to complete
+func waitOperation(t testing.NTB, operation string) error {
+	args := []string{
+		"container", "operations", "wait",
```

Comment:
does wait have a timeout?

Reply:
It doesn't look like it, but we could use a context with a timeout

Reply:
added a timeout context

Reply:
most gke operations take much longer than 1m, some even longer than 5m. like say any upgrade or node pool replacement or repair or even apiserver scale up.

Reply:
bumped the duration